Cannot build gradle classpath

Using neovim version `NVIM v0.1.3-244-g3e0fab6`, Arch Linux, gnome-terminal, I am getting the following errors when opening this Java project: https://github.com/TestFX/TestFX.

```
Messages maintainer: Bram Moolenaar <Bram@vim.org>
"~/dev/testfx/subprojects/testfx-core/src/main/java/org/testfx/matcher/control/TableViewMatchers.java" 97L, 3920C
Error detected while processing function javacomplete#classpath#gradle#BuildClasspathHandler:
line   33:
FAILURE: Build failed with an exception.
Error detected while processing function javacomplete#classpath#gradle#BuildClasspathHandler:
line   33:
* What went wrong:
Error detected while processing function javacomplete#classpath#gradle#BuildClasspathHandler:
line   33:
Failed to notify build listener.
Error detected while processing function javacomplete#classpath#gradle#BuildClasspathHandler:
line   33:
> java.lang.NullPointerException (no error message)
Error detected while processing function javacomplete#classpath#gradle#BuildClasspathHandler:
line   33:
> java.lang.NullPointerException (no error message)
Error detected while processing function javacomplete#classpath#gradle#BuildClasspathHandler:
line   33:
> java.lang.NullPointerException (no error message)
Error detected while processing function javacomplete#classpath#gradle#BuildClasspathHandler:
line   33:
* Try:
Error detected while processing function javacomplete#classpath#gradle#BuildClasspathHandler:
line   33:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output.
Failed to build gradle classpath
```

Can confirm in my case as well (Neovim 0.1.2, OS X 10.11), with the following messages:

```
Error detected while processing function javacomplete#classpath#gradle#BuildClasspathHandler:
line   33:
FAILURE: Build failed with an exception.
Error detected while processing function javacomplete#classpath#gradle#BuildClasspathHandler:
line   33:
* Where:
Error detected while processing function javacomplete#classpath#gradle#BuildClasspathHandler:
line   33:
Script '/Users/vvlasov/.vim/bundle/vim-javacomplete2/classpath.gradle' line: 24
Error detected while processing function javacomplete#classpath#gradle#BuildClasspathHandler:
line   33:
* What went wrong:
Error detected while processing function javacomplete#classpath#gradle#BuildClasspathHandler:
line   33:
Execution failed for task ':classpath'.
```
Plugin works perfectly for Vim (7.4.1525).

--- a/autoload/javacomplete/classpath/gradle.vim
+++ b/autoload/javacomplete/classpath/gradle.vim
@@ -7,7 +7,7 @@ function! javacomplete#classpath#gradle#IfGradle()
     end
   endif
 
-  if (executable('gradle') || executable('./gradlew') || executable('.\gradlew.bat')) && g:JavaComplete_GradlePath != ""
+  if g:JavaComplete_GradlePath != "" && (executable('gradle') || executable(fnamemodify(g:JavaComplete_GradlePath, ':p:h') . (javacomplete#util#IsWindows() ? '\gradlew.bat' : '/gradlew')))
     return 1
   endif
   return 0
@@ -73,9 +73,9 @@ function! s:GenerateClassPath(path, gradle) abort
   if exists("g:JavaComplete_GradleExecutable")
     let gradle = g:JavaComplete_GradleExecutable
   else
-    let gradle = 'gradle'
+    let gradle = fnamemodify(g:JavaComplete_GradlePath, ':p:h') . (javacomplete#util#IsWindows() ? '\gradlew.bat' : '/gradlew')
     if !executable(gradle)
-      let gradle = javacomplete#util#IsWindows() ? '.\gradlew.bat' : './gradlew'
+      let gradle = 'gradle'
     endif
   endif
   call writefile(["allprojects{apply from: '". g:JavaComplete_Home. g:FILE_SEP. "classpath.gradle'}"], s:temporaryGradleFile)

